web/session.go

```diff
@@ -83,6 +85,14 @@ func purgeOldsessionTs() {
 			delete(sessions, key)
 		}
 	}
+
+	for {
+		time.Sleep(time.Second * 60)
+
+		sessionsLock.Lock()
```

Comment:
I would suggest you put the Lock() and then a defer Unlock() inside doPurge itself.